site/docs/v1/tech/lambdas/testing-lambdas.md

```diff
@@ -0,0 +1,380 @@
+
+#### Write The Test
+
+Now you will test that the lambda returns "Goodbye World", which will confirm that the CLI `update` command worked.
+
+Add this code to the `test.js` file.
+
+```js
+{% remote_include 'https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/documentation_snippets/test_2.js' %}
+```
+
+The `login` function calls the FusionAuth Typescript library. It then decodes the JWT response and returns its `message` property.
+
+This `login` function is called by the `tape` function `test`. This test specifies the name of the test, says that it expects exactly one assertion to occur with `plan`, checks that calling `login` returns the property you expect from the lambda updated earlier, and exits. Even if the test fails, the `finally` clause will delete the temporary user created.
+
+Run it with the following command.
+
+```bash
+node test.js
+```
+
+When your code has several tests, and you want a colorful, concise summary, you can use the following instead.
+
+```bash
+node test.js | npx faucet
+```
+
+### Unit Test: Call An External Service
+
+The next test you'll write is a unit test that verifies your lambda locally using a fake mock service and not in FusionAuth. The benefit of this test is that you can test your logic works without needing an external service to be reliable at the time of testing. The danger is that your test might pass locally, but the lambda might fail on FusionAuth due to it running on a different JavaScript environment with different restrictions and configuration.
+
+Let's take an example where you check if users have email addresses from a country sanctioned by the United States, such as North Korea or Cuba. You call the external site `https://issanctioned.example.com` with an email address, and you're told whether or not the domain is banned.
+
+Add this new function to `test.js`. 
+
+```js
+{% remote_include 'https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/documentation_snippets/test_3.js' %}
+```
+
+This is the lambda function that would run on FusionAuth, similar to the "Hello World" function described earlier.
+
+Now add the two tests below. The first test checks if North Korea (`.kp`) is banned and the second if Canada (`.ca`) is allowed.
+
+```js
+{% remote_include 'https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/documentation_snippets/test_4.js' %}
+```
+
+This test function uses `fetchMock` to mock the external service that would be called from the lambda function in FusionAuth. The mocks for the JWT, user, and registration objects are all simple `{}` objects you can pass as parameters to the `populate()` lambda.
+
+Finally, run the tests.
+
+```bash
+node test.js  | npx faucet
+```
```

Comment:
Can we see sample output of the tests (all three)? No need for a screenshot, just copying and pasting the output is fine.